test/source/TestVector.cpp

```diff
@@ -144,6 +144,15 @@ struct testmovable
 	testmovable& operator=(testmovable&&) EA_NOEXCEPT { return *this; }
 };
 
+#if EASTL_MOVE_SEMANTICS_ENABLED
+struct testmovetoself
```

Comment:
Please camel case the struct name.  The class above shouldn't be like that. I'll fix that.

Reply:
Not crazy about the name b/c its actually not testing if its moved to self.  Its really just setting a flag if the copy-assignment operator is called.  I'd either have this class test for self assignment or rename it so the intent is clear.